src/module/cold-storage/usage.ts

```diff
@@ -0,0 +1,190 @@
+import {
+  COLD_STORAGE_HOOK_ADDRESS,
+  COLD_STORAGE_FLASHLOAN_ADDRESS,
+} from './constants'
+import { abi } from './abi'
+import { abi as flashloanAbi } from './flashloanAbi'
+import { Execution } from '../../account'
+import { Account } from '../../account'
+import { moduleTypeIds } from '../types'
+import { SENTINEL_ADDRESS } from '../../common/constants'
+
+type Params = {
+  waitPeriod: number
+}
+
+export const getSetWaitPeriodExecution = ({
+  waitPeriod,
+}: Params): Execution => {
+  return {
+    target: COLD_STORAGE_HOOK_ADDRESS,
+    value: BigInt(0),
+    callData: encodeFunctionData({
+      functionName: 'setWaitPeriod',
+      abi,
+      args: [BigInt(waitPeriod)],
+    }),
+  }
+}
+
+export const getExecutionTime = async ({
+  account,
+  client,
+  executionHash,
+}: {
+  account: Account
+  client: PublicClient
+  executionHash: Hex
+}): Promise<Number | Error> => {
+  try {
+    const executionTimestamp = (await client.readContract({
+      address: COLD_STORAGE_HOOK_ADDRESS,
+      abi,
+      functionName: 'checkHash',
+      args: [account.address, executionHash],
+    })) as Hex
+
+    return Number(executionTimestamp)
+  } catch (err) {
+    console.error(err)
+    throw new Error(
+      `Failed to get execution time for account ${account.address} for execution ${executionHash}`,
+    )
+  }
+}
+
+type RequestTimelockedExecutionParams = {
+  execution: Execution
+  additionalWait: number
+}
+
+export const getRequestTimelockedExecution = ({
+  execution,
+  additionalWait,
+}: RequestTimelockedExecutionParams): Execution => {
+  return {
+    target: COLD_STORAGE_HOOK_ADDRESS,
+    value: BigInt(0),
+    callData: encodeFunctionData({
+      functionName: 'requestTimelockedExecution',
+      abi,
+      args: [execution, additionalWait],
+    }),
+  }
+}
+
+type RequestTimelockedModuleConfigParams = {
+  moduleTypeId: (typeof moduleTypeIds)[keyof typeof moduleTypeIds]
+  module: Address
+  data: Hex
+  isInstall: boolean
+  additionalWait: number
+}
+
+export const getRequestTimelockedModuleConfigExecution = ({
+  moduleTypeId,
+  module,
+  data,
+  isInstall,
+  additionalWait,
+}: RequestTimelockedModuleConfigParams): Execution => {
+  return {
+    target: COLD_STORAGE_HOOK_ADDRESS,
+    value: BigInt(0),
+    callData: encodeFunctionData({
+      functionName: 'requestTimelockedModuleConfig',
+      abi,
+      args: [
+        BigInt(moduleTypeId),
+        module,
+        data,
+        isInstall,
+        BigInt(additionalWait),
+      ],
+    }),
+  }
+}
+
+// -----------------
+// Flashloan actions
+// -----------------
+
+export const getAddAddressExecution = ({
+  addressToAdd,
+}: {
+  addressToAdd: Address
+}): Execution => {
+  return {
+    target: COLD_STORAGE_FLASHLOAN_ADDRESS,
+    value: BigInt(0),
+    callData: encodeFunctionData({
+      functionName: 'addAddress',
+      abi: flashloanAbi,
+      args: [addressToAdd],
+    }),
+  }
+}
+
+export const getRemoveAddressAction = async ({
```

Comment:
I like using action instead of execution but should we port everything over?

Reply:
@kopy-kat Yeah, I like action also more .. I could create a separate PR for that once done with integrating all modules

Reply:
sounds good